src/components/popular-categories/PopularCategories.tsx

```diff
@@ -0,0 +1,80 @@
+import { FC, useCallback, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useNavigate } from 'react-router-dom'
+
+import Box from '@mui/material/Box'
+
+import Loader from '~/components/loader/Loader'
+import ClickableCard from '~/components/clickable-card/ClickableCard'
+import { guestRoutes } from '~/router/constants/guestRoutes'
+import { styles } from '~/components/popular-categories/PopularCategories.styles'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import serviceIcon from '~/assets/img/student-home-page/service_icon.png'
+import ClickableCardList from '~/components/clickable-card-list/ClickableCardList'
+import { CategoryInterface } from '~/types'
+import useBreakpoints from '~/hooks/use-breakpoints'
+
+interface PopularCategoriesProps {
+  title: string
+  description: string
+  items: CategoryInterface[]
+  loading: boolean
+}
+
+const PopularCategories: FC<PopularCategoriesProps> = ({ title, description, items, loading }) => {
+  const { t } = useTranslation()
+  const navigate = useNavigate()
+  const { isDesktop, isTablet, isMobile } = useBreakpoints()
+
+  const itemsToShow = (isDesktop && 9) || (isTablet && 6) || (isMobile && 4) || 9
```

Comment:
Do you need || 9 in the end?

Reply:
I think yes because itemsToShow can return "false | 9 | 6 | 4". And we can't pass false to the method slice. I decided to pass the default value. May you know a better solution?

Reply:
i guess it can`t be false:
xs - sm  = mobile
sm - md = tablet
md and upper = desktop

Reply:
I also think so.
You can see what happens when I leave without || 9
![image](https://user-images.githubusercontent.com/83120263/230016626-af9ca76f-d7a4-4a08-9d4d-eddccc2f0ca3.png)

Reply:
ok